browser/blueprints.py

```diff
@@ -294,46 +273,58 @@ def shortcut(filename):
     request to access a specific data file that has been preloaded to the
     input S3 bucket (ex. http://127.0.0.1:5000/test.npz).
     """
-    rgb = request.args.get('rgb', default='false', type=str)
-    pixel_only = request.args.get('pixel_only', default='false', type=str)
-    label_only = request.args.get('label_only', default='false', type=str)
-
-    settings = {
-        'rgb': bool(distutils.util.strtobool(rgb)),
-        'pixel_only': bool(distutils.util.strtobool(pixel_only)),
-        'label_only': bool(distutils.util.strtobool(label_only))
-    }
-
-    if is_trk_file(filename):
-        filetype = 'track'
-        title = 'Tracking Tool'
-
-    elif is_npz_file(filename):
-        filetype = 'zstack'
-        title = 'Z-Stack Tool'
-
-    else:
-        # TODO: render an error template instead of JSON.
-        error = {
-            'error': 'invalid file extension: {}'.format(
-                os.path.splitext(filename)[-1])
-        }
-        return jsonify(error), 400
-
+    settings = make_settings(filename)
     return render_template(
         'tool.html',
-        filetype=filetype,
-        title=title,
-        filename=filename,
         settings=settings)
 
 
 def get_edit(project):
     """Factory for Edit objects"""
-    filename = project.filename
+    filename = project.path
     if is_npz_file(filename):
         return ZStackEdit(project)
     elif is_trk_file(filename):
         # don't use RGB mode with track files
         return TrackEdit(project)
     return BaseEdit(project)
+
+
+def make_settings(filename):
+    """Returns a dictionary of settings to send to the front-end."""
+    folders = re.split('__', filename)
```

Comment:
Is this different than `filename.split('__')`?

Reply:
I don't think so. Change it to reduce the number of imports?

Reply:
actually `re` is also used in line 211 
```
path = re.sub('__', '/', filename)
```
but we could use `filename.replace('__', '/')` instead to get rid of all `re` calls

Reply:
Doesn't matter to me, `re` is the standard library. Just wanted to make sure as I think the previous version used `str.split()`.